Tasks/FtpUpload/ftpuploadtask.ts

```diff
@@ -0,0 +1,254 @@
+/// <reference path="../../definitions/vsts-task-lib.d.ts" />
+/// <reference path="../../definitions/node.d.ts" />
+/// <reference path="../../definitions/Q.d.ts" />
+
+
+import fs = require('fs');
+import os = require('os');
+import path = require('path');
+import tl = require('vsts-task-lib/task');
+import url = require('url');
+
+
+var win = os.type().match(/^Win/);
+tl.debug('win: ' + win);
+
+var repoRoot: string = tl.getVariable('build.sourcesDirectory');
+function makeAbsolute(normalizedPath: string): string {
+    tl.debug('makeAbsolute:' + normalizedPath);
+
+    var result = normalizedPath;
+    if (!path.isAbsolute(normalizedPath)) {
+        result = path.join(repoRoot, normalizedPath);
+        tl.debug('Relative file path: ' + normalizedPath + ' resolving to: ' + result);
+    }
+    return result;
+}
+
+function failTask(message: string) {
+    tl.setResult(tl.TaskResult.Failed, message);
+}
+
+// server endpoint
+var serverEndpoint = tl.getInput('serverEndpoint', true);
+var serverEndpointUrl : url.Url = url.parse(tl.getEndpointUrl(serverEndpoint, false));
+
+var serverEndpointAuth = tl.getEndpointAuthorization(serverEndpoint, false);
+var username = serverEndpointAuth['parameters']['username'];
+var password = serverEndpointAuth['parameters']['password'];
+
+// the root location which will be uploaded from
+var rootFolder: string = makeAbsolute(path.normalize(tl.getPathInput('rootFolder', true).trim()));
+if (!tl.exist(rootFolder)) {
+    failTask('The specified root folder: ' + rootFolder + ' does not exist.');
```

Comment:
If the root folder path is long, this message will be more readable as:
`'The specified root folder does not exist: ' + rootFolder`